### PR TITLE
Remove module docstring titles

### LINKED ISSUE #10160: Follow PEP 257 docstring conventions
[PEP 257](https://www.python.org/dev/peps/pep-0257/) establishes conventions for docstrings, in particular:

>Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description. The summary line may be used by automatic indexing tools; it is important that it fits on one line and is separated from the rest of the docstring by a blank line.

Sphinx currently violates that convention by formatting its module docstrings as follows:

```
"""
    sphinx.builders
    ~~~~~~~~~~~~~~~

    Builder superclass for all builders.

    :copyright: Copyright 2007-2022 by the Sphinx team, see AUTHORS.
    :license: BSD, see LICENSE for details.
"""
```

Note that the summary is not in the first line but after a heading and a blank line. This is problematic because when e.g. browsing the Sphinx API via [Pydoctor](https://github.com/twisted/pydoctor), Pydoctor just shows the headings (since it assumes the PEP 257 convention):

image

So it would be great if Sphinx could follow PEP 257 by removing such headings from its module docstrings.

### PR DESCRIPTION
**Pull request by AA-Turner**
Closes #10160 

Alternative to #10208

@tk0miya -- I had been working on an expanded version of your PR, you beat me to it!

Over and above #10208, this conforms module docstrings to [PEP 257](https://python.github.io/peps/pep-0257/#multi-line-docstrings) by moving the summary line next to the opening quotes, dedents the text, ensures summary lines are less than 80 characters, and also removes redundant copyright & licencing information (see [this post](https://www.linuxfoundation.org/blog/copyright-notices-in-open-source-software-projects/) for more infomation on why this is OK.) 

I did keep the copyright notices where a different licence has been used, though.

A

### CODE REVIEW

File: sphinx/ext/napoleon/docstring.py [hunk 1]
```diff
@@ -1,13 +1,7 @@
 """
-    sphinx.ext.napoleon.docstring
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Classes for docstring parsing and formatting.
```
Comment (tk0miya) [inline sphinx/ext/napoleon/docstring.py]: This file contains empty lines before and after the first sentence.

File: sphinx/ext/napoleon/iterators.py [hunk 2]
```diff
@@ -1,13 +1,7 @@
 """
-    sphinx.ext.napoleon.iterators
-    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+A collection of helpful iterators.
```
Comment (tk0miya) [inline sphinx/ext/napoleon/iterators.py]: This is also.

File: tests/test_ext_napoleon.py [hunk 3]
```diff
@@ -1,12 +1,5 @@
-"""
-    test_napoleon
-    ~~~~~~~~~~~~~
-
-    Tests for :mod:`sphinx.ext.napoleon.__init__` module.
-
+"""Tests for :mod:`sphinx.ext.napoleon.__init__` module.
```
Comment (tk0miya) [inline tests/test_ext_napoleon.py]: This contains an empty line.

File: tests/test_ext_napoleon_docstring.py [hunk 4]
```diff
@@ -1,12 +1,5 @@
-"""
-    test_napoleon_docstring
-    ~~~~~~~~~~~~~~~~~~~~~~~
-
-    Tests for :mod:`sphinx.ext.napoleon.docstring` module.
-
+"""Tests for :mod:`sphinx.ext.napoleon.docstring` module.
```
Comment (tk0miya) [inline tests/test_ext_napoleon_docstring.py]: This is also.

File: tests/test_ext_napoleon_iterators.py [hunk 5]
```diff
@@ -1,12 +1,5 @@
-"""
-    test_napoleon_iterators
-    ~~~~~~~~~~~~~~~~~~~~~~~
-
-    Tests for :mod:`sphinx.ext.napoleon.iterators` module.
-
+"""Tests for :mod:`sphinx.ext.napoleon.iterators` module.
```
Comment (tk0miya) [inline tests/test_ext_napoleon_iterators.py]: This is also.

File: tests/test_intl.py [hunk 6]
```diff
@@ -1,12 +1,6 @@
-"""
-    test_intl
-    ~~~~~~~~~
-
-    Test message patching for internationalization purposes.  Runs the text
-    builder in the test root.
+"""Test message patching for internationalization purposes.
```
Comment (tk0miya) [inline tests/test_intl.py:2]: This is also.

**Review by tk0miya — COMMENTED** [review]
LGTM with nits.

**Comment by AA-Turner** [comment]
@tk0miya I've fixed all the ones you raised.

A

File: tests/test_intl.py (hunk 6 above)
Comment (AA-Turner) [reply tests/test_intl.py:2]: I think this is fine -- the blank line is as there's a longer description on line 3:

https://github.com/sphinx-doc/sphinx/blob/f05d455313d2a521723f8f99f30b86a86fce1d4c/tests/test_intl.py#L1-L4

A

Comment (tk0miya) [reply tests/test_intl.py:2]: Oops, you're right. withdrawing.

**Comment by tk0miya** [comment]
Thank you for your update. Merging now. Good work!

**Pull request merged**